src/main/java/fr/openent/viescolaire/service/impl/DefaultPeriodeService.java

```diff
@@ -188,23 +188,34 @@ public void handle(Either<String, JsonArray> stringEntryEither) {
                                             @Override
                                             public void handle(Either<String, List<String>> stringListEither) {
                                                 if(stringListEither.isRight()) {
-                                                    final String[] idClasses = stringListEither.right().getValue().toArray(new String[0]);
-                                                    getPeriodesGroupe(idEtablissement, idClasses,
-                                                            new Handler<Either<String, JsonArray>>() {
-                                                                @Override
-                                                                public void handle(Either<String, JsonArray> stringJsonArrayEither) {
-                                                                    if (stringJsonArrayEither.isRight()) {
-                                                                        for (Object o : stringJsonArrayEither.right().getValue()) {
-                                                                            JsonObject groupePeriode = (JsonObject) o;
-                                                                            groupePeriode.put("id_groupe", idGroupe);
+                                                    //if the groupe is not linked to classes
+                                                    if (stringListEither.right().getValue().isEmpty())  {
+                                                        groupHandler.handle(new Either.Right<String, JsonArray>(new JsonArray()));
+                                                    } else {
+                                                        final String[] idClasses = stringListEither.right().getValue().toArray(new String[0]);
+                                                        getPeriodesGroupe(idEtablissement, idClasses,
+                                                                new Handler<Either<String, JsonArray>>() {
+                                                                    @Override
+                                                                    public void handle (Either<String, JsonArray> stringJsonArrayEither) {
```

Comment:
à minima, si tu ne veux pas utiliser les promises:
```suggestion
                                                                    getPeriodesGroupe(idEtablissement, idClasses, event -> {
                                                                    ...
                                                                    } 
```